cmd/healthinfo_linux.go

```diff
@@ -38,31 +38,29 @@ func getLocalOsInfo(ctx context.Context, r *http.Request) madmin.ServerOsInfo {
 		addr = GetLocalPeer(globalEndpoints)
 	}
 
-	info, err := host.InfoWithContext(ctx)
+	srvrOsInfo := madmin.ServerOsInfo{Addr: addr}
+	var err error
+
+	srvrOsInfo.Info, err = host.InfoWithContext(ctx)
 	if err != nil {
 		return madmin.ServerOsInfo{
 			Addr:  addr,
 			Error: fmt.Sprintf("info: %v", err),
 		}
 	}
 
-	sensors, err := host.SensorsTemperaturesWithContext(ctx)
+	srvrOsInfo.Sensors, err = host.SensorsTemperaturesWithContext(ctx)
 	if err != nil {
-		return madmin.ServerOsInfo{
-			Addr:  addr,
-			Error: fmt.Sprintf("sensors-temp: %v", err),
+		// Set error only when it's not of WARNINGS type
+		if _, isWarning := err.(*host.Warnings); !isWarning {
```

Comment:
Instead of dropping the information would it make sense to add it as a field?

Reply:
From the perspective of subnet health report/analyzer it won't be very useful. We won't be writing any checks on the warnings or showing them to the user (on `subnet health` UI).

In general we are trying to minimize the size of the health report and include only the data really required for subnet health.

Maybe we can add the `warnings` field if required in future, if we feel the need to make use of the warning messages in the UI or health checks.